torch/csrc/autograd/saved_variable_hooks.h

```diff
@@ -10,4 +10,10 @@ struct TORCH_API SavedVariableHooks {
   virtual ~SavedVariableHooks() = default;
 };
 
+struct TORCH_API DefaultSavedVariableHooks {
```

Comment:
How do we intend to extend this to a potential C++ api in the future?
It feels a little weird having this interface define only one function to be shared between the (maybe you intend to add reset_hooks as well?).
I feel like its slightly less useful to have this common class (vs the SavedVariableHooks) case because there's no place in the code where you need the polymorphism. The get default engine should already handle that for you right?

Reply:
It is a good point that right now, I don't need it for polymorphism.
We can always rewrite this code when we need it.